src/textual/_cells.py

```diff
@@ -1,10 +1,44 @@
 from typing import Callable
 
-__all__ = ["cell_len"]
+from textual.expand_tabs import get_tab_widths
+
+__all__ = ["cell_len", "cell_width_to_column_index"]
 
 
 cell_len: Callable[[str], int]
 try:
     from rich.cells import cached_cell_len as cell_len
 except ImportError:
     from rich.cells import cell_len
+
+
+def cell_width_to_column_index(line: str, cell_width: int, tab_width: int) -> int:
```

Comment:
Could a small `lru_cache` work here? i.e. will this be called frequently with the same values?

Reply:
An LRU cache would mean that if you move the cursor up and down over the same character on a line, without editing that line, we'd hit the cache.

Also if you hit click the same offset on the same line multiple times, it'd be faster.

Could be wrong as I'm just going off a feeling here, but I don't feel like we'd get many cache hits on a small cache here, and a big cache would have a relatively large memory cost (relatively speaking) since it's caching full lines.

Reply:
Fair enough. If its not called frequently then there is probably little benefit. It just jumped out as a "pure function".